packages/atomic/src/components/search/atomic-result-children/atomic-result-children.ts

```diff
@@ -0,0 +1,353 @@
+      getTemplateHasError: () => this.templateHasError,
+      setResultTemplateRegistered: (value: boolean) => {
+        this.resultTemplateRegistered = value;
+      },
+      setTemplateHasError: (value: boolean) => {
+        this.templateHasError = value;
+      },
+      bindings: this.bindings,
+    });
+  }
+
+  private resolveChildTemplates = (event: ChildTemplatesContextEvent) => {
+    event.preventDefault();
+    const provider =
+      this.itemTemplateProvider ||
+      this.childTemplatesContextController.itemTemplateProvider;
+    event.detail(provider ?? undefined);
+  };
+
+  private get foldedResultList(): FoldedResultList | null {
+    return this.foldedItemListContextController.foldedItemList;
+  }
+
+  private get result(): FoldedResult | null {
+    return this.resultContextController.item as FoldedResult | null;
+  }
+
+  private get displayConfig(): DisplayConfig | null {
+    return this.displayConfigContextController.config;
+  }
+
+  private get effectiveItemTemplateProvider(): ResultTemplateProvider | null {
+    return (
+      this.itemTemplateProvider ||
+      this.childTemplatesContextController.itemTemplateProvider
+    );
+  }
+
+  private get collection() {
+    if (!this.foldedResultListState || !this.result) {
+      return undefined;
+    }
+    return this.foldedResultListState.results.find((r) => {
+      return r.result.uniqueId === this.result?.result.uniqueId;
+    });
+  }
+
+  willUpdate() {
+    if (this.foldedResultList && !this.foldedResultListUnsubscriber) {
+      this.foldedResultListUnsubscriber = this.foldedResultList.subscribe(
+        () => {
+          this.foldedResultListState = this.foldedResultList!.state;
+          if (!this.initialChildren && this.collection) {
+            this.initialChildren = this.collection.children;
+          }
+        }
+      );
+      this.foldedResultListState = this.foldedResultList.state;
+    }
+
+    if (!this.initialChildren && this.collection) {
+      this.initialChildren = this.collection.children;
+    }
+  }
+
+  private loadFullCollection() {
+    this.loadedFullCollection = true;
+    this.dispatchEvent(
+      buildCustomEvent('atomic/loadCollection', this.collection)
+    );
+  }
+
+  private toggleShowInitialChildren = () => {
+    if (!this.foldedResultList || !this.result) {
+      return;
+    }
+
+    if (this.showInitialChildren) {
+      this.foldedResultList.logShowMoreFoldedResults(this.result.result);
+    } else {
+      this.foldedResultList.logShowLessFoldedResults();
+    }
+
+    this.showInitialChildren = !this.showInitialChildren;
+  };
+
+  private renderChild(child: FoldedResult, isLast: boolean) {
+    const content = this.effectiveItemTemplateProvider?.getTemplateContent(
+      child.result
+    );
+
+    if (!content) {
+      return nothing;
+    }
+
+    const key =
+      child.result.uniqueId +
+      child.children.map((c) => c.result.uniqueId).join('');
```

Comment:
Nitpick: If we want to be completely equivalent to the Stencil version, we'd join with `','` here.